src/Text/Pandoc/Writers/ICML.hs

```diff
@@ -270,20 +270,30 @@ escapeColons = Text.concatMap $ \x -> case x of
   ':' -> "%3a"
   _   -> Text.singleton x
 
+-- | figure out the link destination for a given URL
+-- | HyperlinkURLDestination with more than one colon crashes CS6
+makeDest :: Text -> Doc Text
+makeDest x = case Text.uncons x of
+    Just ('#', _) -> literal $ "HyperlinkTextDestination/"<>escapeColons (escapeStringForXML x)
+    _             -> literal $ "HyperlinkURLDestination/"<>escapeColons (escapeStringForXML x)
```

Comment:
welcome to the world of Haskell! :-)
nit-pick, but I'd probably write it more like this:

```suggestion
makeDest :: Text -> Doc Text
makeDest txt = literal $
  if "#" `Text.isPrefixOf` txt
  then "HyperlinkTextDestination/" <> escTxt
  else "HyperlinkURLDestination/" <> escTxt
  where
    escTxt = escapeColons $ escapeStringForXML txt
```

Reply:
I like that syntax - thanks!  Much cleaner (and it explains nicely how `where` works)